### PR TITLE
Branch to correct Validation.Rmd and adjust_seeding_ages  errors.

### PR DESCRIPTION
**Pull request by ceeb950**
Function adjust_seeding_ages intermittently fails and I have checked … in the code for this failure and run the function again when this happens. Validation.Rmd is now running without the size error.

### CODE REVIEW

**Comment by ceeb950** [comment]
@richfitz @OJWatson @giovannic - please review; errors corrected - https://github.com/mrc-ide/hypatia/pull/44

File: vignettes/Validation.Rmd [hunk 1]
```diff
@@ -24,23 +24,39 @@ iso3c <- "ATG"
...
-pop$n <- round(pop$n/10)
+pop$n <- as.integer(pop$n / 10)
 
 # Create our equivalent parameters
+R0 <- 2
 time_period <- 200
+tt_contact_matrix <- 0
 
-parameters <-  hypatia::get_parameters(
+parameters <- hypatia:::get_parameters(
```
Comment (richfitz) [inline vignettes/Validation.Rmd]: triple colon is not needed here (or above on l. 24) as `get_parameters` is an exported function

```suggestion
parameters <- hypatia::get_parameters(
```

File: R/variables.R [hunk 2]
```diff
@@ -104,65 +109,61 @@ create_variables <- function(pop, parameters) {
...
-  # position of iv to be swapped out
+  # # position of iv to be swapped out
   to_distribute <- tail(seq_along(iv), length(ages))
 
   # position of iv to be swapped in
   to_swap <- vector()
+
   for (i in seq_along(unique(ages))) {
+    tsi1 <- list()
+    tsi2 <- list()
```
Comment (richfitz) [inline R/variables.R]: these two lines can be deleted as the variables are immediately overwritten

```suggestion
```

**Review by giovannic — APPROVED** [review]

File: R/variables.R [hunk 3]
```diff
@@ -104,65 +109,61 @@ create_variables <- function(pop, parameters) {
...
+    tsi1 <- which(iv == unique(ages)[i])
+    tsi2 <- head(tsi1, sum(ages == unique(ages)[i]))
```
Comment (richfitz) [inline R/variables.R]: You might move `unique(ages)` up outside of this loop as it is constant and used twice. Longer term it should be possible to use something like `findInterval` to split this more efficiently but let's worry about that later if this flashes up in profiling

File: vignettes/Validation.Rmd [hunk 4]
```diff
@@ -24,23 +24,39 @@ iso3c <- "ATG"
...
+
+Run our simulation - default timesteps = 200
+
+```{r}
+
 df_1 <- hypatia::run_simulation(pop, parameters)
 
+```
+
+PLOT RESULTS
```
Comment (richfitz) [inline vignettes/Validation.Rmd]: You can use `##` for headings or `*` around words for emphasis

**Review by richfitz — CHANGES_REQUESTED** [review]
Nice job getting to the bottom of the problem here. A few suggestions and minor things to tidy here

**Comment by ceeb950** [comment]
@richfitz @OJWatson - please review; rich's review changes made - https://github.com/mrc-ide/hypatia/pull/44

**Review by richfitz — APPROVED** [review]

**Comment by ceeb950** [comment]
Closes #44 and #23.

**Comment by ceeb950** [comment]
Closes #23 and closes #44

**Pull request merged**